crates/hir_def/src/nameres.rs

```diff
@@ -105,6 +105,8 @@ pub struct DefMap {
     ///
     /// (the primary purpose is to resolve derive helpers and fetch a proc-macros name)
     exported_proc_macros: FxHashMap<MacroDefId, ProcMacroDef>,
+    /// Side table storing the modules of macro definitions.
+    macro_modules: FxHashMap<MacroDefId, LocalModuleId>,
```

Comment:
We could merge these two technically, but I feel like we will waste more memory that way, as the majority of macros are declarative, so storing a lot of `Option<ProcMacroDef>::None` seems wasteful.